editor/editor_translation_parser.h

```diff
@@ -0,0 +1,73 @@
+/*  editor_translation_parser.h                                          */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef EDITOR_TRANSLATION_PARSER_H
+#define EDITOR_TRANSLATION_PARSER_H
+
+#include "core/error_list.h"
+#include "core/reference.h"
+
+class EditorTranslationParserPlugin : public Reference {
+	GDCLASS(EditorTranslationParserPlugin, Reference);
+
+protected:
+	static void _bind_methods();
+
+public:
+	virtual Error parse_file(const String &p_path, Vector<String> *r_extracted_strings);
+	virtual void parse_text(const String &p_text, Vector<String> *r_extracted_strings);
+	virtual void get_recognized_extensions(List<String> *r_extensions) const;
+};
+
+class EditorTranslationParser {
+	static EditorTranslationParser *singleton;
+
+public:
+	enum ParserType {
+		STANDARD, // GDScript, CSharp, ...
+		CUSTOM // User-defined parser plugins. This will override standard parsers if the same extension type is defined.
+	};
```

Comment:
I'm not sure that the standard/custom distinction is necessary, it doesn't seem to be used by other plugin systems which have both built-in plugins and user-defined ones.

Though indeed `ResourceLoader::_load` currently checks all plugins and takes the first valid one, so I'm not sure how conflicts are handled. So maybe the approach you used it a good idea.

Reply:
Actually it was just to make it easier to write add_parser() and remove_parser().
I didn't want to write add_standard_parser(), remove_standard_parser(),  add_custom_parser(), remove_custom_parser() so I thought it would be better to just adjust with enum parameter.
But I could change it if you think not having the enum makes it clearer code.

Reply:
No I'm ok with the enum, I just raised a question on whether we want to distinguish standard and custom parsers, or treat them all the same.